core/store/src/db.rs

```diff
@@ -49,22 +49,66 @@ impl Into<io::Error> for DBError {
 #[derive(PartialEq, Debug, Copy, Clone, EnumIter, BorshDeserialize, BorshSerialize, Hash, Eq)]
 pub enum DBCol {
```

Comment:
I think we should provide rust like interface to the database.
```rust
get<DBCol::ColBlock>(arg: CryptoHash) -> near_primitives::block_header::BlockHeader
```

Reply:
@matklad What do you think?

Reply:
that's an interesting idea - but it might be a larger scope than this PR intended.

Maybe let's start a zulip thread about it (how the storage interface should look like).
It might be interesting, as AFAIK Evgeny is thinking about changing the underlying storage to something else, so having an interface might help